### PR TITLE
VACMS-3577: content release job status block

### PR DESCRIPTION
**Pull request by acrollet**
## Description

See #3577 

## Screenshots

chrome-capture (2)

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change. 
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by indytechcook** [comment]
Can you make this a PR against https://github.com/department-of-veterans-affairs/va.gov-cms/pull/3864?  Perhaps we can merge it since we can't merge 3864 until after the deploy today.

**Comment by acrollet** [comment]
> Can you make this a PR against #3864? Perhaps we can merge it since we can't merge 3864 until after the deploy today.

will do

File: docroot/modules/custom/va_gov_build_trigger/src/Controller/ContentReleaseStatusBlockController.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  /**
+   * Get content_release_status_block content.
+   *
+   * @return string
+   *   Rendered block.
+   */
+  public function getBlock() {
+    $block = $this->pluginManagerBlock->createInstance('content_release_status_block');
+    $output = render($block->build());
+    return new Response($output);
```
Comment (indytechcook) [inline docroot/modules/custom/va_gov_build_trigger/src/Controller/ContentReleaseStatusBlockController.php:39]:
```suggestion
    return Response::create($output);
```

As a general rule, try to use factory methods to create classes.

File: docroot/modules/custom/va_gov_build_trigger/src/Plugin/Block/ContentReleaseStatusBlock.php [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+   */
+  private function getCommandRunnerJob() {
+    return $this->database->select('advancedqueue', 'aq')
+      ->condition('aq.queue_id', 'command_runner')
+      ->fields('aq')
+      ->range(0, 1)
+      ->orderBy('available', 'DESC')
+      ->execute()
+      ->fetchObject();
+  }
```
Comment (indytechcook) [inline docroot/modules/custom/va_gov_build_trigger/src/Plugin/Block/ContentReleaseStatusBlock.php:253]:
```suggestion
  private function getCommandRunnerJob() : ?stdClass {
    return $this->database->select('advancedqueue', 'aq')
      ->condition('aq.queue_id', 'command_runner')
      ->fields('aq')
      ->range(0, 1)
      ->orderBy('available', 'DESC')
      ->execute()
      ->fetchObject() ?? NULL;
  }
```

Not really required just a thought

Comment (acrollet) [reply docroot/modules/custom/va_gov_build_trigger/src/Plugin/Block/ContentReleaseStatusBlock.php:253]: ah cool, works for me :-)

**Comment by acrollet** [comment]
> Can you make this a PR against #3864? Perhaps we can merge it since we can't merge 3864 until after the deploy today.

@indytechcook here's the PR: https://github.com/acrollet/va.gov-cms/pull/1

**Comment by acrollet** [comment]
@indytechcook I've updated the PR with the items we discussed. I didn't switch to use the entity query manager because although queues are entities, jobs are not. (I also didn't find any API for querying jobs in advancedqueue, it just claims jobs one at a time...)

**Comment by indytechcook** [comment]
> @indytechcook I've updated the PR with the items we discussed. I didn't switch to use the entity query manager because although queues are entities, jobs are not. (I also didn't find any API for querying jobs in advancedqueue, it just claims jobs one at a time...)

I stand corrected, though it does feel like a miss on their part.  I think it would be easier to read if we used `Drupal\advancedqueue\Job` with something like this https://www.drupal.org/docs/7/api/database-api/fetching-into-a-custom-class (if possible;f or just loading the class manually.  There is no other way than using `new Job` to create the object so that's ok.

**Review by indytechcook — APPROVED** [review]

**Pull request merged**